src/server/worldsettings.cpp

```diff
@@ -0,0 +1,197 @@
+/*
+Minetest
+Copyright (C) 2010-2013 celeron55, Perttu Ahola <celeron55@gmail.com>
+Copyright (C) 2013-2020 Minetest core developers & community
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "worldsettings.h"
+#include "filesys.h"
+#include "log.h"
+
+#define MAP_BACKEND_KEY "backend"
+#define DEFAULT_BACKEND "sqlite3"
+#define MAP_READONLY_BACKEND_KEY "readonly_backend"
+#define AUTH_BACKEND_KEY "auth_backend"
+#define PLAYER_BACKEND_KEY "player_backend"
+
+#if USE_REDIS
+#define REDIS_ADDR_KEY "redis_address"
+#define REDIS_HASH_KEY "redis_hash"
+#define REDIS_PASSWORD_KEY "redis_password"
+#define REDIS_PORT_KEY "redis_port"
+#endif
```

Comment:
There should be a generic solution instead of hardcoding the names here + in function names.

Reply:
a generic solution to store config keys ? you mean a enum string instead ?

Reply:
currently solution is better than before but maybe not so optimal (because we have config keys everywhere in MT, thanks to g_settings object :(